src/lib.rs

```diff
@@ -96,439 +100,371 @@ fn validate_container(container: &apicore::Container) -> bool {
 #[cfg(test)]
 mod tests {
     use super::*;
+    use rstest::rstest;
 
-    #[test]
-    fn accept_pod_when_all_ephemeral_containers_are_not_privileged_test() -> Result<()> {
-        let result = validate_pod(&apicore::PodSpec {
-            ephemeral_containers: Some(vec![
-                apicore::EphemeralContainer {
+    #[rstest]
+    #[case::accept_pod_when_all_ephemeral_containers_are_not_privileged_test(apicore::PodSpec {
```

Comment:
I think we could increase the readibility by creating helper functions and use them inside the case, such as:

```rust
    #[case::accept_pod_when_all_ephemeral_containers_are_not_privileged_test(
        pod_with_all_ephemeral_containers_not_privileged(),
        Settings::default(),
        Ok(true),
        "Pod with no privileged ephemeral container should be accepted by the validator"
    )]
```

We could even have factory functions to dry up even more the test code:

```rust
    #[case::accept_pod_when_all_ephemeral_containers_are_not_privileged_test(
        pod_with_ephemeral_containers_factory(&[false, false, false]),
        Settings::default(),
        Ok(true),
        "Pod with no privileged ephemeral container should be accepted by the validator"
    )]
```
Wich could create the `apicore::PodSpec` struct for the case.

Wdyt?

Reply:
Yeah, that's a good idea. I'll do it